### DISCUSSION TITLE
Accept pinned GitHub Action updates from branches

### DISCUSSION THREAD

**Discussion by matfax** [renovatebot/renovate#23338] [Suggest an Idea]
### Tell us more.

At the moment, renovatebot accepts tags in the comment line after the hash for reference (#23136).
```
uses: user/action@3db72ac7a2798f125adcdf60fa7711df5c22879ffbf896c99de905275a7aaf02 # v1.0.0
```

This, however, doesn't seem to work if the origin of the hashed commit is a branch. In matfax/mutapath#317, I tried `renovate: tag=release`, `renovate: branch=release`, and `release` but renovatebot still fetched the sha from the default branch, not the release branch.

Some GitHub action use distribution branches, or duplicate branches for shortened versions (e.g. `v1` and `v1.0`) which continue to be updated within the same version prefix, e.g. when a new fix `v1.0.1` is released.

Possibly related to this: #22320

**Comment by rarkins** [comment]
Same answer as #22320

**Reply by matfax** [reply]
I'm a bit confused by this discussion, including the answer. If it only uses `github-tags`, how does it fetch the latest hash from the default/master/main branch? From an outside perspective, the bot would only need to change the branch name. As a distinction from tag comments, a prefix like `# branch=name` could be introduced to use `git-refs`.

**Reply by rarkins** [reply]
It doesn't fetch latest hash, it fetches hashes from GitHub's git tags endpoint: https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28

Only if there's no tag/version does it fall back to a regular git ref check.
